pkg/authenticator/config/config.go

```diff
@@ -28,12 +29,37 @@ const (
 	DefaultClientCertPath = "/etc/conjur/ssl/client.pem"
 	DefaultTokenFilePath  = "/run/conjur/access-token"
 
+	DefaultConjurVersion = "5"
+
 	// DefaultTokenRefreshTimeout is the default time the system waits to reauthenticate on error
 	DefaultTokenRefreshTimeout = 6 * time.Minute
 )
 
+// NewFromEnvFunc defines the method for creating a new config from the env
+type NewFromEnvFunc func(Options) (*Config, error)
+// ReadSSLCertFunc defines the method for reading an SSL Certificate from the env
+type ReadSSLCertFunc func() ([]byte, error)
+// NewUserNameFunc defines the method for parsing a username from the env
+type NewUserNameFunc func(string) (*Username, error)
+// Options contains the functional options needed for NewFromEnv
+type Options struct {
+	ReadSSLCert ReadSSLCertFunc
+	NewUserName NewUserNameFunc
+}
+
+// DefaultConfigOptions implement the methods defined within config.go
+var DefaultConfigOptions = Options{
+	readSSLCert,
+	NewUsername,
+}
+
+// DefaultNewFromEnv returns NewFromEnv with the default functional options
+func DefaultNewFromEnv() (*Config, error) {
```

Comment:
Normally, the prefix for Golang constructors is `New` so if a change is needed, it should contain methods similar to `NewFromEnv` and `NewFromEnvWithOptions`.